### PR TITLE
Migrate InteractiveMarkerDisplay

### LINKED ISSUE #98: Migrate InteractiveMarkerDisplay

### PR DESCRIPTION
**Pull request by jacobperron**
Closes #98 

Depends on https://github.com/ros2/ros2/pull/770

Some changes since ROS 1 based on discussion in https://github.com/ros-visualization/interactive_markers/issues/42:
* The feedback publisher is now maintained by the interactive marker client (not RViz)
* Since clients and servers now communicate with a mix of services and topics, the ROS topic property should be replaced with custom property for interactive marker server discovery.

### CODE REVIEW

**Comment by jacobperron** [comment]
Opening for visibility. This isn't quite ready for review as there are some things I'd still like to do:

* Add a custom property object for selecting interactive marker servers to connect to (as eluded to in the description).
* Review code for areas that can be refactored for the better
  * So far I've done the minimum to get the code to compile and run without crashing, but there may be places in the code that might be ripe for updating.
* Add tests. Visual tests would be nice.

For those interested in trying out the display, you can check out this branch and clone the interactive_markers package to your workspace: https://github.com/ros-visualization/interactive_markers/tree/ros2.

I've ported the demos too. You can find them here: https://github.com/jacobperron/visualization_tutorials/tree/jacob/ros2

**Comment by jacobperron** [comment]
Planning to add some tests still, but the rest is ready for review.

File: rviz_default_plugins/include/rviz_default_plugins/displays/interactive_markers/interactive_marker_control.hpp [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+   * The direction vector of the axis of rotation during a mouse
+   * drag, relative to the reference frame.
+   * Computed on mouse down event.
+  */
+  Ogre::Vector3 rotation_axis_;
+
+  /// The center of rotation during a mouse drag, relative to the control frame.
+  /**
+   * Computed on mouse down event.
+  */
```
Comment (wjwwood) [inline rviz_default_plugins/include/rviz_default_plugins/displays/interactive_markers/interactive_marker_control.hpp]: nitpick: alignment of `*` is wrong here.

**Review by wjwwood — APPROVED** [review]

File: rviz_default_plugins/include/rviz_default_plugins/displays/interactive_markers/interactive_marker_control.hpp (hunk 1 above)
Comment (jacobperron) [reply rviz_default_plugins/include/rviz_default_plugins/displays/interactive_markers/interactive_marker_control.hpp]: 5dfa654

**Review by wjwwood — APPROVED** [review]

**Comment by jacobperron** [comment]
I've been working on tests, but it looks like we should add/extend mock objects.
@wjwwood If it's alright with you I'll merge this (pending CI) and follow-up with improving test coverage.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8098/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4097/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6598/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7972/)

**Comment by wjwwood** [comment]
Sounds good to me.

**Pull request merged**